Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -213,48 +242,34 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate {
 
 extension SearchViewController: UITableViewDataSource {
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
-
-        if let currentSection = SearchListSection(rawValue: indexPath.section) {
-            switch currentSection {
-            case .SearchSuggestions:
-                cell.textLabel?.text = searchSuggestions[indexPath.row]
-                cell.detailTextLabel?.text = nil
-                cell.imageView?.image = nil
-                cell.isAccessibilityElement = false
-                cell.accessibilityLabel = nil
-            case .BookmarksAndHistory:
-                if let site = data[indexPath.row] as? Site {
-                    cell.textLabel?.text = site.title
-                    cell.detailTextLabel?.text = site.url
-                    if let img = site.icon? {
-                        let imgUrl = NSURL(string: img.url)
-                        cell.imageView?.sd_setImageWithURL(imgUrl, placeholderImage: self.profile.favicons.defaultIcon)
-                    } else {
-                        cell.imageView?.image = self.profile.favicons.defaultIcon
-                    }
-                    cell.isAccessibilityElement = false
-                    cell.accessibilityLabel = nil
+        switch SearchListSection(rawValue: indexPath.section)! {
```

Comment:
What you are saying about implicitly unwrapped optionals in classes also really worries me. Swift provides us with tools to make sure that code is *proven correct* when it comes to memory management.

One of number one crashers in Android are NullPointerExceptions. Why is that? Because dealing with state is one of the most difficult problems when writing software.

Swift gives us tools that force us to write code that is correct. If you use those features then there cannot be a case of "where we forgot to assign them" - simply because that is not possible.